pkg/addon/manifests/deployment.yaml

```diff
@@ -0,0 +1,50 @@
+kind: Deployment
+apiVersion: apps/v1
+metadata:
+  name: klusterlet-addon-workmgr
+  namespace: {{ .Namespace }}
+  labels:
+    app: work-manager
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: work-manager
+  template:
+    metadata:
+      labels:
+        app: work-manager
+    spec:
+      serviceAccountName: klusterlet-addon-workmgr-sa
+      securityContext:
+        runAsNonRoot: true
+      volumes:
+      - name: hub-config
+        secret:
+          secretName: {{ .KubeConfigSecret }}
+      containers:
+      - name: klusterlet-addon-workmgr
+        image: {{ .Image }}
+        args:
+          - "/agent"
+          - "--hub-kubeconfig=/var/run/hub/kubeconfig"
+          - "--cluster-name={{ .ClusterName }}"
+          - "--port=4443"
+          - "--agent-address=klusterlet-addon-workmgr.open-cluster-management-agent.svc"
```

Comment:
there are 3 cases here:
1. --agent-address: use svc in local-cluster
2. --agent-route: route on OCP
3. --agent-servic:  loadbalancer

refer: https://github.com/open-cluster-management/klusterlet-addon-operator/blob/020356206ba0e26153c8bdb54953012f6dab829e/helm-charts/work-manager/templates/deployment.yaml#L55